Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs

```diff
@@ -3428,28 +3433,59 @@ private void SwitchFrame(ElementInfo EI)
             }
         }
 
-        public IWebElement LocateElementByLocators(ObservableList<ElementLocator> Locators, ElementInfo currentPOMElementInfo,bool IscallfromFriendlyLocator = false)
+        public IWebElement LocateElementByLocators(ElementInfo currentPOMElementInfo,bool iscallfromFriendlyLocator=false, POMExecutionUtils POMExecutionUtils=null)
         {
             IWebElement elem = null;
-            foreach (ElementLocator locator in Locators)
+            foreach (ElementLocator locator in currentPOMElementInfo.Locators)
             {
                 locator.StatusError = string.Empty;
                 locator.LocateStatus = ElementLocator.eLocateStatus.Pending;
             }
 
-            foreach (ElementLocator locator in Locators.Where(x => x.Active == true).ToList())
+            foreach (ElementLocator locator in currentPOMElementInfo.Locators.Where(x => x.Active == true).ToList())
             {
-                if (!locator.IsAutoLearned)
+                List<FriendlyLocatorElement> friendlyLocatorElementlist = new List<FriendlyLocatorElement>();
+                if (locator.EnableFriendlyLocator && !iscallfromFriendlyLocator)
                 {
-                    elem = LocateElementIfNotAutoLeared(locator);
+                    IWebElement targetElement = null;
+                   
+                    foreach (ElementLocator FLocator in currentPOMElementInfo.FriendlyLocators.Where(x=>x.Active == true).ToList())
+                    {
+                        if(!FLocator.IsAutoLearned)
+                        {
+                            ElementLocator evaluatedLocator = FLocator.CreateInstance() as ElementLocator;
+                            ValueExpression VE = new ValueExpression(this.Environment, this.BusinessFlow);
+                            FLocator.LocateValue = VE.Calculate(evaluatedLocator.LocateValue);
+                        }
+
+                        if (FLocator.LocateBy == eLocateBy.POMElement)
+                        {
+                            ElementInfo ReferancePOMElementInfo = POMExecutionUtils.GetFriendlyElementInfo(new Guid(FLocator.LocateValue));
+
+                            targetElement = LocateElementByLocators(ReferancePOMElementInfo, true);
+                        }
+                        else
+                        {
+                            targetElement = LocateElementByLocator(FLocator);
+                        }
+                        if(targetElement!=null)
+                        {
+                            FriendlyLocatorElement friendlyLocatorElement = new FriendlyLocatorElement();
+                            friendlyLocatorElement.position = (FLocator.Position == ePosition.LeftOf ? "left" : (FLocator.Position == ePosition.RightOf ? "right" : ( FLocator.Position == ePosition.Below ? "below" : ( FLocator.Position == ePosition.Above ? "above" : (FLocator.Position == ePosition.Near ? "near": "")))));
```

Comment:
In the enum use the values like left instead of ePosition.LeftOf

and then just to tostring here instead of all these conditions

Reply:
Done